git/remote.py

```diff
@@ -694,10 +715,26 @@ def push(self, refspec=None, progress=None, **kwargs):
 
         :param refspec: see 'fetch' method
         :param progress:
-            Instance of type RemoteProgress allowing the caller to receive
-            progress information until the method returns.
             If None, progress information will be discarded
 
+            No further progress information is returned after push returns.
+
+            A function (callable) that is called with the progress infomation:
+
+            progress( op_code, cur_count, max_count=None, message='' )
+
+            op_code is a bit mask of values defined in git.RemoteProgress
```

Comment:
Instead of repeating the [existing documentation](http://gitpython.readthedocs.io/en/stable/reference.html?highlight=RemoteProgress#git.remote.RemoteProgress.update) in place, could you link to it ? It's fine to link to readthedocs, but certainly is possible to link to an API class *locally* as well. Unfortunately I don't know by heart how.

As you sometimes provide more information than is given in the existing one, you could consider manually merging it there.